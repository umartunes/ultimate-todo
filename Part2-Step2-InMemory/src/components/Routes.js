import React, { Component, Fragment } from 'react'
import { Switch, Route } from 'react-router-dom'
import AddTodoPage from './AddTodoPage';
import Dashboard from './Dashboard';
import EditTodoPage from './EditTodoPage';


class Routes extends Component {

	render() {
		return (
			<Fragment>
				<Switch>
					<Route  exact={true} path="/" component={Dashboard} />
					<Route exact={true} path="/create" component={AddTodoPage }/>
					<Route exact={true} path="/edit/:id" component={EditTodoPage }/>
				</Switch>
			</Fragment>
		)
	}

}

export default Routes;
