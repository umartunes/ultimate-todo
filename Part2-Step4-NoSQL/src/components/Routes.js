import React, { Component, Fragment } from 'react'
import { Switch, Route } from 'react-router-dom'


import Dashboard from './Dashboard';
import AddTodo from './AddTodo';
import EditTodo from './EditTodo';
class Routes extends Component {

	render() {
		return (
			<Fragment>
				<Switch>
					<Route exact={true} path="/" component={Dashboard} />
					<Route exact={true} path="/create" component={AddTodo} />
					<Route exact={true} path="/edit/:id" component={EditTodo} />
				</Switch>
			</Fragment>
		)
	}

}

export default Routes;
