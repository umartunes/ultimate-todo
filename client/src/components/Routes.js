import React, { Component, Fragment } from 'react'
import { Switch, Route } from 'react-router-dom'
import PrivateRoute from './privateRoute'
import Login from './Login'
import Dashboard from './Dashboard';
import AddTodo from './AddTodo';
class Routes extends Component {

	render() {
		return (
			<Fragment>
				<Switch>
					<PrivateRoute  exact={true} path="/" component={Dashboard} />
					{/* <Route exact={true} path ="/dashboard" component={Dashboard}/> */}
					<PrivateRoute exact={true} path="/create" component={AddTodo }/>

				</Switch>
			</Fragment>
		)
	}

}

export default Routes;
