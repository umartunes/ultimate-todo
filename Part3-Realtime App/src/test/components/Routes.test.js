import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Routes from '../../components/Dashboard';




test('Should Render Routes Correctly',()=>{
    
    const renderer = new ReactShallowRenderer();
    renderer.render(<Routes/>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
})