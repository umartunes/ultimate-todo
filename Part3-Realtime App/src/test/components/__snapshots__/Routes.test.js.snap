// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Routes Correctly 1`] = `
<Unknown>
  <SideNave />
  <Header />
  <Section />
</Unknown>
`;
